keyboard/src/lib.rs

```diff
@@ -1,11 +1,433 @@
 #![no_std]
 
-static CODES: [u8; 59] = *b"??1234567890-=??qwertyuiop[]\n?asdfghjkl;'`?\\zxcvbnm,./?*? ?";
+pub enum Action {
+    Pressed,
+    Released,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Key {
+    KA,
```

Comment:
The Rust compiler wasn't happy with `0`, `1`, etc. as plain numbers. I updated it to use plain letters/words for most of the codes, and then these keyboard (not keypad) number codes are prefixed with `Keyboard`. Also included a comment before them to note the difference between the keyboard number keys and the keypad number keys.